### DISCUSSION TITLE
getWorldFormModelMatrix  and getWorldFromModelNormalMatrix()	 have bug ！

### DISCUSSION THREAD

**Discussion by Haidiye00** [google/filament#7082] [Q&A]
After my testing, I found there is a bug in the getWorldFormModelMatrix()   API 
this is test code.
![Cjg9dk5zfF](https://github.com/google/filament/assets/132528822/5f294952-ee0b-4bcb-a7d7-75821c492c36)
This is the correct rendering.
![cstrU7fuuD](https://github.com/google/filament/assets/132528822/11383a81-4e5b-4c17-afe5-42f5a41015f0)
This is abnormal rendering.
![W1rbJQ1Nc8](https://github.com/google/filament/assets/132528822/3f08f6f6-7677-46ab-bcdf-a834f7650c2f)
And also there is a bug in the getWorldFromModelNormalMatrix()   API 
This is the correct rendering.
![7sQacxbrPI](https://github.com/google/filament/assets/132528822/76cb20ec-e728-455a-ba4b-66ea8f48dc99)
This is abnormal rendering.
![R2KcWYGks3](https://github.com/google/filament/assets/132528822/ca8081d2-3c4f-4cb1-8425-be03be8e17e5)

**Comment by romainguy** [comment]
Those are the exact matrices used by default to transform objects. Note that you shouldn't need to use them since the default behavior is to transform objects in world coordinates.

**Reply by romainguy** [reply]
For reference here's what Filament does by default:

```
    mat4 transform = getWorldFromModelMatrix();
    vec3 position = getPosition().xyz;
    return mulMat4x4Float3(transform, position);
```
